Mathlib/CategoryTheory/Products/Basic.lean

```diff
@@ -352,4 +353,25 @@ def functorProdFunctorEquiv : (A ⥤ B) × (A ⥤ C) ≌ A ⥤ B × C :=
     counitIso := functorProdFunctorEquivCounitIso A B C, }
 #align category_theory.functor_prod_functor_equiv CategoryTheory.functorProdFunctorEquiv
 
+section Opposite
+
+open Opposite
+
+/-- The equivalence between the opposite of a product and the product of the opposites. -/
+@[simps]
+def prodOp : (C × D)ᵒᵖ ≌ Cᵒᵖ × Dᵒᵖ where
+  functor :=
+    { obj := λ X => ⟨op X.unop.1, op X.unop.2⟩,
+      map := λ f => ⟨f.unop.1.op, f.unop.2.op⟩ }
+  inverse :=
+    {
+      obj := λ ⟨X,Y⟩ => op ⟨X.unop, Y.unop⟩,
+      map := λ ⟨f,g⟩ => op ⟨f.unop, g.unop⟩ }
+  unitIso := by aesop_cat
+  counitIso := by aesop_cat
+  functor_unitIso_comp := λ ⟨⟨X,Y⟩⟩ => by
+    simp; rw[Category.id_comp (𝟙 (op X)),Category.id_comp (𝟙 (op Y))]; simp
```

Comment:
```suggestion
  functor_unitIso_comp := fun ⟨X, Y⟩ => by
    dsimp
    ext <;> simpa using Category.id_comp _
```
Since the transition to Lean4, `fun` is usually preferred to `λ`. In my suggestion, nonterminal simps are removed.

Reply:
How come `simp` doesn't close the goal directly and we need to use `id_comp` manually?

Reply:
Also, what is it that allows matching `⟨X, Y⟩` directly rather than first unpacking the `Opposite` structure? Is it a general feature since `Opposite` has only one field?